airbyte-integrations/connectors/destination-clickhouse/src/main/kotlin/io/airbyte/integrations/destination/clickhouse/client/ClickhouseSqlGenerator.kt

```diff
@@ -51,6 +53,19 @@ class ClickhouseSqlGenerator(
         return this
     }
 
+    private fun isValidVersionColumnType(airbyteType: AirbyteType): Boolean {
+        // Must be of an integer type or of type Date/DateTime/DateTime64
+        val validTypes =
```

Comment:
Consider making this a constant/file level field to avoid creating it each time this method is called.